### PR TITLE
Fixed #22569 -- Allowed ModelAdmin.lookup_allowed() to respect get_list_filter().

### PR DESCRIPTION
**Pull request by DrMeers**
https://code.djangoproject.com/ticket/22569

### CODE REVIEW

**Comment by DrMeers** [comment]
Actually I wonder whether we should just make the `request` parameter mandatory since it isn't fully backwards-compatible if you've written a custom `lookup_allowed` method that doesn't accept `request`.

File: docs/releases/2.0.txt [hunk 1]
```diff
@@ -41,7 +41,12 @@ Minor features
 :mod:`django.contrib.admin`
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* ...
+* ``ModelAdmin.lookup_allowed`` now accepts a ``request`` parameter to ensure
+  it respects ``get_list_filter``. The parameter is optional to maintain
+  backwards compatibility.
+* A new ``ModelAdmin.get_changelist_instance`` method creates new override
```
Comment (timgraham) [inline docs/releases/2.0.txt]: Other than using this in the new test, I don't see how it's linked to the rest of the PR. I think it should have its own ticket, tests, docs, and PR. That'll help simplify this one.

**Comment by DrMeers** [comment]
Agreed on both counts, thanks @timgraham

**Comment by timgraham** [comment]
Simon, do you want to continue this?

**Comment by timgraham** [comment]
Closing due to inactivity.

**Comment by DrMeers** [comment]
Sorry @timgraham , I really do want to continue this but am not sure if/when I'll find the time :(

**Pull request closed without merging**